app/src/main/java/org/oppia/android/app/utility/ClickableAreasImage.kt

```diff
@@ -122,16 +122,18 @@ class ClickableAreasImage(
    * Toggles whether the clickable region corresponding to the provided [clickableArea] is visible
    * and available to be clicked.
    */
-  fun toggleRegion(clickableArea: ImageWithRegions.LabeledRegion) {
+  fun toggleRegion(clickableArea: ImageWithRegions.LabeledRegion, view: View?) {
     resetRegionSelectionViews()
     listener.onClickableAreaTouched(
       NamedRegionClickedEvent(
         clickableArea.label,
         clickableArea.contentDescription
       )
     )
-    val newView = createSelectableView(clickableArea)
-    newView.setBackgroundResource(R.drawable.selected_region_background)
+
+    view?.setBackgroundResource(R.drawable.selected_region_background) ?: createSelectableView(
+      clickableArea
+    ).setBackgroundResource(R.drawable.selected_region_background)
```

Comment:
```suggestion
    val affectedView = view ?: createSelectableView(clickableArea)
    affectedVoew.setBackgroundResource(R.drawable.selected_region_background)
```

I find the current code a bit hard to follow, so here's a more concise alternative.

Reply:
Updated